src/nanaimo/instruments/jlink/__init__.py

```diff
@@ -17,59 +17,114 @@
+                               default='default')
+        arguments.add_argument('--serial-interface',
+                               help='The serial interface to use (e.g. swd, jtag)',
+                               default='swd')
+        arguments.add_argument('--reset-delay-millis',
+                               default=400,
+                               help='Milliseconds to wait after reset before starting to load the new image.')
+
+    def on_construct_command(self, arguments: nanaimo.Namespace, inout_artifacts: nanaimo.Artifacts) -> str:
+        """
+        Construct a command to upload (eke "flash") a firmware to a target device using Segger's JLink
+        Commander program with the assumption that a Segger debug probe like the JLink is attached to the
+        system.
+        +----------------+---------------------------------------+--------------------------------------------------+
+        | **Artifacts**                                                                                             |
+        |                                                                                                           |
+        +----------------+---------------------------------------+--------------------------------------------------+
+        | key            | type                                  | Notes                                            |
+        +================+=======================================+==================================================+
+        | ``scriptfile`` | Optional[tempfile.NamedTemporaryFile] | A temporary file containing the jlink script to. |
+        |                |                                       | execute.                                         |
+        +----------------+---------------------------------------+--------------------------------------------------+
+        .. invisible-code-block: python
+
+            from nanaimo.fixtures import FixtureManager
+            from nanaimo import Namespace, Arguments, Artifacts
+            from nanaimo.instruments import jlink
+
+            import os
+            import asyncio
+
+            loop = asyncio.get_event_loop()
+            manager = FixtureManager(loop=loop)
+
+            cmd = jlink.ProgramUploader(manager)
+
+            ns = Namespace()
+            setattr(ns, '{}_hexfile'.format(jlink.ProgramUploader.get_argument_prefix()), 'foo.hex')
+            setattr(ns, '{}_device'.format(jlink.ProgramUploader.get_argument_prefix()), 'S32K188')
+            artifacts = Artifacts()
+
+            command = cmd.on_construct_command(ns, artifacts)
+            assert command.find('-CommanderScript') > 0
+
+        """
+        tmpfile = tempfile.NamedTemporaryFile(mode='w')
+        # keep around for as long as the command exists.
+        setattr(inout_artifacts, 'scriptfile', tmpfile)
+
+        # poor man's templating
+        expanded_script_file = self.script.format(
```

Comment:
I think it'd be helpful to let an argument override this default generated script file, for users of the fixture that already have their own JLinkCommand script or need to generate one differently. Then you can skip the template open-write below, if that argument was present.

Reply:
Updated. Let me know if the new diff works better.

Reply:
👍